### DISCUSSION TITLE
Blazor WASM components in Razorpages

### DISCUSSION THREAD

**Discussion by Wulfheart** [dotnet/aspnetcore#26421] [General] [answered]
I have a existing application using Razorpages. I found possibilities to integrate server-rendered/server-prerendered Razor components, but I would like to use some client-side Razor components in my application.
Is this possible? If yes, how can I achieve this?

**Comment by SteveSandersonMS** [comment, marked as answer]
Yes, this is something we've added as a built-in feature with .NET 5 😄 

Please see https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-5-release-candidate-1/#blazor-webassembly-prerendering. Note that instead of `render-mode="WebAssemblyPrerendered"`, you can use `render-mode="WebAssembly"` if you just need the component to execute on WebAssembly without being prerendered.

**Reply by Wulfheart** [reply]
Thank you really much for your response. 
Is it different to register compared to Blazor Server?

**Reply by SteveSandersonMS** [reply]
There are various differences - please see the setup steps in the blog post for details.

**Reply by Wulfheart** [reply]
Ah, I just saw it is only a Release Candidate. Any ETA for Release?

**Reply by SteveSandersonMS** [reply]
.NET 5 will be released in November. I don't think there's an exact date within November just yet.

**Reply by Wulfheart** [reply]
Thank you. Then I will use it in the next version. Release Candidates are not that good in production.
